src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -407,6 +408,29 @@ void VM_Version::initialize() {
     UsePopCountInstruction = true;
   }
 
+  if (UseBranchProtection == nullptr || strcmp(UseBranchProtection, "none") == 0) {
+    _rop_protection = false;
+  } else if (strcmp(UseBranchProtection, "standard") == 0) {
+    _rop_protection = false;
+    // Enable PAC if this code has been built with branch-protection and the CPU/OS supports it.
+#ifdef __ARM_FEATURE_PAC_DEFAULT
+    if (_features & CPU_PACA) {
+      _rop_protection = true;
+    }
+#endif
+  } else if (strcmp(UseBranchProtection, "pac-ret") == 0) {
+    _rop_protection = true;
+#ifdef __ARM_FEATURE_PAC_DEFAULT
+    if (!(_features & CPU_PACA)) {
+      warning("UseROPProtection specified, but not supported on this CPU.");
+    }
+#else
+    warning("UseROPProtection specified, but not supported in the VM.");
+#endif
```

Comment:
If we issue these warnings should `_rop_protection` still be set true?

Reply:
As per this conversation: https://github.com/openjdk/jdk/pull/6334#discussion_r791722292

The idea was, the user is explicitly asking for asking for pac-ret so we should honour that. Whereas standard would only enable what is supported for that system.

Reply:
But we can't honour that because it is not supported. Further, the suggestion in the referenced discussion seemed to be based on the assumption that doing so would be harmless because it is NOP based, but you have indicated that may not be the case and so it may actually lead to a crash!

Reply:
Before I change anything - @theRealAph you had an opinion here too...

Reply:
Given that the implementation has now changed so much that it's no longer NOP based, I'll go with @dholmes-ora .
One other thing, though: it might be better to say here "but this VM was built without ROP-protection support." That's more informative, IMO.

Reply:
Ok, I'll fix up as suggested.

The beginning part of that message needs fixing too - UseROPProtection is no longer the name of the flag. I'll switch to:
"ROP-protection specified, but this VM was built without ROP-protection support."

Reply:
And this change will keep ROP protection enabled if we fall into the "this VM was built without ROP-protection support.". In that case we'll be protecting generated code, but the VM itself won't be protected. This will run without crashing.

Reply:
> And this change will keep ROP protection enabled if we fall into the "this VM was built without ROP-protection support.". In that case we'll be protecting generated code, but the VM itself won't be protected. This will run without crashing.

That's perfect.

Reply:
Okay, that makes sense. Thanks.